django/contrib/gis/gdal/raster/source.py

```diff
@@ -73,6 +73,13 @@ def __init__(self, ds_input, write=False):
 
         # If input is a valid file path, try setting file as source.
         if isinstance(ds_input, str):
+            if (
+                not ds_input.startswith(VSI_FILESYSTEM_BASE_PATH) and
```

Comment:
Should we replace `and` with `or`?
what about a file with a path like `'/vsimem/nonexistent.tif'`

Reply:
`capi.open_ds()` will handle them.